### PR TITLE
Use environment invalidation for cuda_configure

### LINKED ISSUE #4848: "bazel clean" will undo "./configure" when source configured to build for GPU.
### Environment info

Operating System:
ubuntu 14.04

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
cuda 8.0.44, cudnn 5.5

If installed from binary pip package, provide:
No

If installed from source, provide 
1. The commit hash (`git rev-parse HEAD`)
   c8d4896e3231c3ac32f174fd0af051867645fbb5
2. The output of `bazel version`
   Build label: 0.3.1
   Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
   Build time: Fri Jul 29 09:09:52 2016 (1469783392)
   Build timestamp: 1469783392
   Build timestamp as int: 1469783392
### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

./configure   # configure with GPU support)
bazel test -c opt --config=cuda tensorflow/core:common_runtime_direct_session_test # this one should pass
bazel clean
bazel test -c opt --config=cuda tensorflow/core:common_runtime_direct_session_test  # this one fails, complaining no GPU support configured.
### Logs or other output that would be helpful

(If logs are large, please upload as attachment or provide link).
java.lang.RuntimeException: Unrecoverable error while evaluating node 'CONFIGURATION_FRAGMENT:com.google.devtools.build.lib.skyframe.ConfigurationFragmentValue$ConfigurationFragmentKey@ae
3629bd' (requested by nodes 'CONFIGURATION_COLLECTION:com.google.devtools.build.lib.skyframe.ConfigurationCollectionValue$ConfigurationCollectionKey@1374ec1e', 'CONFIGURATION_FRAGMENT:com
.google.devtools.build.lib.skyframe.ConfigurationFragmentValue$ConfigurationFragmentKey@3686b55', 'CONFIGURATION_FRAGMENT:com.google.devtools.build.lib.skyframe.ConfigurationFragmentValue
$ConfigurationFragmentKey@a93d9174')
        at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:1070)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:474)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

David, Damien, looks like bazel clean undid some output of "./configure" at head.
Any idea where things went wrong?

### PR DESCRIPTION
**Pull request by damienmg**
This removes the needs for clean --expunge in configure :) but
request the incoming bazel 0.4.5.

This should fix #4848

### CODE REVIEW

**Comment by damienmg** [comment]
We will cut a new release soonish and that should include the necessary change to that, of course this make all the presubmit fails right now because it uses a non existent bazel feature

**Comment by davidzchen** [comment]
Awesome! \o/

**Comment by vrv** [comment]
Nice, thanks.  this will have to wait for our CI to upgrade.  assigned gunan/yifei

**Comment by yifeif** [comment]
Thanks! Preparing a CL, will upgrade once 0.4.5 is out.

**Comment by dandelionmane** [comment]
(Continues to be stalled while waiting on bazel 0.4.5)

**Comment by damienmg** [comment]
I don't think @gunan wanted to close that PR when merging his change, now this PR should be ready to merge.

**Comment by gunan** [comment]
Sorry, yeah I think that was auto closed due to "Fix ...".

Now that we are updated, Damien could you sync and resolve conflicts?

**Comment by damienmg** [comment]
Merged. I don't understand how I can make the windows CMake test fails :p

**Comment by gunan** [comment]
win1 may be in a bad state.
Rerunning here:
http://ci.tensorflow.org/job/tensorflow-pr-win-cmake-py/1672/console

File: third_party/gpus/cuda_configure.bzl [hunk 1]
```diff
@@ -824,10 +824,20 @@ def _cuda_autoconf_impl(repository_ctx):
     _create_cuda_repository(repository_ctx)
 
 
+
 cuda_configure = repository_rule(
     implementation = _cuda_autoconf_impl,
-    local = True,
+    environ = [
+        _GCC_HOST_COMPILER_PATH,
+        "TF_NEED_CUDA",
```
Comment (gunan) [inline third_party/gpus/cuda_configure.bzl]: This one modifies the behaviour in configure script, but does anything else depend on it?

Comment (gunan) [inline third_party/gpus/cuda_configure.bzl]: I found one usage here in this file.

**Review by gunan — APPROVED** [review]

**Comment by gunan** [comment]
Jenkins, test this please.

File: third_party/gpus/cuda_configure.bzl (hunk 1 above)
Comment (damienmg) [inline third_party/gpus/cuda_configure.bzl]: I added all the environment variable used in the cuda_configure.bzl script.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by theyonibomber** [comment]
This broke the ability to build for me, please advise... (See #8619)

**Pull request merged**